### PR TITLE
Aggressively group prod and dev dependencies for NPM

### PR DESCRIPTION
**Pull request by abdulapopoola**
Use wildcard matching to aggregate prod and dev updates into two separate groups

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -50,6 +50,20 @@ updates:
...
       time: "16:00"
+    groups:
+      dependencies:
+        patterns:
+          - "*yarn*"
+          - "*arborist*"
+          - "*nock*"
+          - "*npm*"
+          - "*semver*"
+          - "detect-indent"
```
Comment (landongrindheim) [inline .github/dependabot.yml:61]: 🔎 I don't know how likely it is that new dependencies will be introduced and catch us by surprise, but I think I'd be more comfortable with the patterns being the [list of dependencies](https://github.com/dependabot/dependabot-core/blob/cd0af0cf66dc9488c30bf07203cf644e6c271e1d/npm_and_yarn/helpers/package.json#L12-L19) as they curently exist (i.e. hard-coding them). That would make the kind of errors that _I'm_ susceptible to harder to introduce 😅  What do you think of that?

If that's something you're agreeable to, I think 👇 would do the trick 😄 
```suggestion
          - "@dependabot/yarn-lib"
          - "@npmcli/arborist"
          - "detect-indent"
          - "nock"
          - "npm"
          - "@pnpm/lockfile-file"
          - "@pnpm/dependency-path"
          - "semver"
```

**Review by landongrindheim — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (abdulapopoola) [reply .github/dependabot.yml]: Good suggestion; agree with being explicit over implicit. Thank you!

File: .github/dependabot.yml [hunk 2]
```diff
@@ -50,6 +50,22 @@ updates:
...
+          - "@npmcli/arborist"
+          - "detect-indent"
+          - "nock"
+          - "npm"
+          - "@pnpm/lockfile-file"
+          - "@pnpm/dependency-path"
+          - "semver"
+      dev-dependencies:
+        patterns:
+          - "*eslint*"
```
Comment (abdulapopoola) [inline .github/dependabot.yml:66]: @landongrindheim would you want the same 'explicit' configuration to apply to dev-dependencies too? I can make the update too.

Comment (landongrindheim) [reply .github/dependabot.yml:66]: I'm less concerned with the dev dependencies 😄

**Review by deivid-rodriguez — APPROVED** [review]
Happy to try it!

**Pull request merged**